File: final_dataset_A/generated_code/chatgpt/java/ImportCustomerRecordsXML-chatgpt.java
import java.io.File;
import javax.xml.parsers.DocumentBuilder;
import javax.xml.parsers.DocumentBuilderFactory;
import org.w3c.dom.*;

public class CustomerImporter {

    public static void main(String[] args) {
        try {
            File xmlFile = new File("customers.xml");

            DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
            DocumentBuilder builder = factory.newDocumentBuilder();
            Document document = builder.parse(xmlFile);

            document.getDocumentElement().normalize();

            NodeList customers = document.getElementsByTagName("customer");

            for (int i = 0; i < customers.getLength(); i++) {
                Node node = customers.item(i);

                if (node.getNodeType() == Node.ELEMENT_NODE) {
                    Element customer = (Element) node;

                    String id = customer.getAttribute("id");
                    String name = customer.getElementsByTagName("name").item(0).getTextContent();
                    String email = customer.getElementsByTagName("email").item(0).getTextContent();

                    System.out.println("Customer ID: " + id);
                    System.out.println("Name: " + name);
                    System.out.println("Email: " + email);
                    System.out.println("-------------------------");
                }
            }

        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}

File: generated-code-groupA-updated/generated-code-groupA-updated/chatgpt/python-chatgpt-A/import_customer_records_xml-chatgpt.py
import xml.etree.ElementTree as ET

def import_customers(xml_file):
    tree = ET.parse(xml_file)
    root = tree.getroot()

    for customer in root.findall("customer"):
        customer_id = customer.get("id")
        name = customer.find("name").text
        email = customer.find("email").text

        print(f"Customer ID: {customer_id}")
        print(f"Name: {name}")
        print(f"Email: {email}")
        print("-" * 30)

if __name__ == "__main__":
    import_customers("customers.xml")